File: prodcon.go
package main

import (
"container/list"
"fmt"
"math/rand"
"time"
)

var c = make(chan int, 3)

func Producer() {
	fmt.Printf("start prod\n")
	for i := 0; i < 10; i++ {
		r := rand.Intn(1000000)
		c<-r
		fmt.Printf("Producer added %d\n",r)
		//time.Sleep(time.Duration(rand.Intn(10)) * time.Millisecond)
	}
}
func Consumer1() {
	fmt.Printf("start con 1\n")
	for i := 0; i < 10; i++ {
		r  := <-c
		fmt.Printf("consumer 1 removed %d\n",r)
		time.Sleep(time.Duration(rand.Intn(300)) * time.Millisecond)
	}
}
func Consumer2() {
	fmt.Printf("start con 2\n")
	for i := 0; i < 10; i++ {
		r := <-c
		fmt.Printf("consumer 2 removed %d\n",r)
		time.Sleep(time.Duration(rand.Intn(300)) * time.Millisecond)
	}
}
func main() {
	rand.Seed( time.Now().UTC().UnixNano())
	go Producer()
	go Consumer1()
	go Consumer2()
	time.Sleep(10000 * time.Millisecond)
}

File: prodcon.py
#prodcon.py

import threading
import Queue
from random import randint
import time


q = Queue.Queue(0);

def prod():
	while 1:
		rand = randint(1,100000)
		q.put(rand)
		print str(rand) + " was added to queue\n"
		time.sleep(1)
def con1():
	time.sleep(3)
	while 1:
		print str(q.get()) + " was removed from queue by consumer 1\n"
		time.sleep(randint(1,3))

def con2():
	time.sleep(3)
	while 1:
		print str(q.get()) + " was removed from queue by consumer 2\n"
		time.sleep(randint(1,3))

prod = threading.Thread(target=prod)
con1 = threading.Thread(target=con1)
con2 = threading.Thread(target=con2)
prod.start()
con1.start()
con2.start()
prod.join()
con1.join()
con2.join()